### DISCUSSION TITLE
Search with a prefix, like ABC{searchTerms}, possible?

### DISCUSSION THREAD

**Discussion by ptrdvrs** [microsoft-search/pnp-modern-search#3881] [Q&A]
In the past I had a working solution for searching with a prefix, but now it isn't working anymore (for some people). For example I've SharePoint libraries and most of them starts in the Title with ABC and then 3 numbers (ABC123 or ABC253).
In the searchbox I want to enter "123" and find the library. 

In the past I had (for example): ABC{searchTerms} AND contentclass:"STS_List_DocumentLibrary"
Now that's working anymore (for most people, strange, for some it still working, is userrelated because moves with user to other machine)
Also (ABC{searchTerms}) AND contentclass:"STS_List_DocumentLibrary"
isn't working.

Using the last version, just upgraded to be sure.
Is there a way how to get this done?

At the same time the * within the ({searchTerm}*) stops working.

**Comment by kasperbolarsen** [comment]
Title:ABC{searchTerms} contentclass:"STS_List_DocumentLibrary" should do the trick

**Reply by ptrdvrs** [reply]
Thanks, that is a good solution, much better than I had first. For me it works! But unfortunately for other users it does not, same as the issue as before. I hoped another solution would work.... 
<img width="620" alt="JSR" src="https://github.com/user-attachments/assets/866d9357-c007-48eb-9162-89cdf64475a5">

If I put something before of after {searchTerms} for some users the results always none.

**Reply by kasperbolarsen** [reply]
This sounds very much some a reindexing issue, but just to be absolutely sure, can you have her test it on a classic search center or in the ootb Microsoft search , like this 
![image](https://github.com/user-attachments/assets/ff94602a-8001-4daf-a5a8-edfc3343b152)

Next step would be to trigger a reindexing of the site collection(s) that are used in you setup

**Reply by ptrdvrs** [reply]
Thanks for the tip to put te search in the MS search. We did that, and she found the site.
We have several pnp-search on that site (page) and all of them are working accept when we add something direct 'next' (without space) 
to {searchTerms} in the query. On another site collection, the issue is the same and on that one (is smaller in size) I did a reindex of the site, no luck. A new site with a copy of a part of the original site has the same issue. Nothing changed in the users, same rights.
Thinking of some regional settings maybe?
![Schermafbeelding 2024-08-01 150718](https://github.com/user-attachments/assets/e96e7353-f908-4c23-a7fd-4713ed9ef3e7)

**Reply by kasperbolarsen** [reply]
Not likely.  Have you tried to inspect the query in the browser dev tools while she is doing the search?

**Reply by ptrdvrs** [reply]
Yes, we get both the same in de Dev Tools. But I get the result en the other user not. When I send the link (of the library) they have fully access to the files.
QueryTemplate: "Title:KP20180{SearchTerms} AND contentclass:\"STS_List_DocumentLibrary\""
Querytext: "390"
![image](https://github.com/user-attachments/assets/7bfd7e55-2146-4abc-b6b5-6078f27fb4b4)

**Reply by kasperbolarsen** [reply]
Any chance you can set up a classic page with the classic search web parts and using the same search query/setup? If that shows the same issue you now have an excellent case to put in a service request ticket with MS. (They will reject the case if any not-ms solution, like PnP , is involved)

**Reply by ptrdvrs** [reply]
That's a good tip. I've created a classic page with search results and a new search setting to send searchquery to new classic page. Now wait till Monday.... (end users are all gone now). I'll post an update when I know something!

**Comment by ptrdvrs** [comment]
@kasperbolarsen The end user tested both pages (old/plain MS search and modern search) and the clssic page gives a good result, but the modern search doesn't. I've done the same and I get both a result. The query is the same (except the Searchboxquery vs searchterms). A support ticket to Mircosoft won't work. Screenshot of the query (edit mode), results my account, results end user. Thanks in advance.
<img width="1271" alt="Edit search" src="https://github.com/user-attachments/assets/9442dedf-df17-473e-8ab5-669db2ed826e">
<img width="648" alt="Search" src="https://github.com/user-attachments/assets/c3d73bc4-8a82-4735-9e8f-45a3ce6a662b">
<img width="953" alt="Working search" src="https://github.com/user-attachments/assets/982f61ba-f965-4944-b786-36f065a445c4">
